tests/cloud/conftest.py

```diff
@@ -0,0 +1,170 @@
+
+def pytest_generate_tests(metafunc):  
+    """
+    pytest_generate_tests hook is used to create our own input parametrization,
+    It generates all the test cases of different combination of input parameters which are read from the json file,
+    and passed to each testing script module.
+    -----------
+    Ref: https://docs.pytest.org/en/7.3.x/how-to/parametrize.html
+    """
+    json_data = None
+    json_file  = os.path.join(ROOT_DIR,"tests","HL_testing_input.json")
+    with open(json_file,'r') as file:
+        json_data =  json.load(file)
+    print("\n**************JSON data***************\n\n",json_data)
+
+    if ('model_name' in metafunc.fixturenames and 'num_cores' in metafunc.fixturenames and 'prompt' in metafunc.fixturenames
+        and 'prompts_txt_file_path' in metafunc.fixturenames
+        and 'aic_enable_depth_first' in metafunc.fixturenames and 'mos' in metafunc.fixturenames and 'cache_dir' in metafunc.fixturenames 
+        and 'hf_token' in metafunc.fixturenames and 'batch_size' in metafunc.fixturenames and 'prompt_len' in metafunc.fixturenames 
+        and 'ctx_len' in metafunc.fixturenames and 'mxfp6' in metafunc.fixturenames and 'mxint8' in metafunc.fixturenames
+        and 'device_group' in metafunc.fixturenames):
+
+        metafunc.parametrize("model_name", json_data['model_name'])
+        metafunc.parametrize("num_cores", json_data['num_cores'])
+        metafunc.parametrize("prompt",json_data['prompt'])
+        metafunc.parametrize("prompts_txt_file_path",json_data['prompts_txt_file_path'])
+        metafunc.parametrize("aic_enable_depth_first",json_data['aic_enable_depth_first'])
+        metafunc.parametrize("mos",json_data['mos'])
+        metafunc.parametrize("cache_dir",[Constants.CACHE_DIR])
+        metafunc.parametrize("hf_token",json_data['hf_token'])
+        metafunc.parametrize("batch_size",json_data['batch_size'])
+        metafunc.parametrize("prompt_len",json_data['prompt_len'])
+        metafunc.parametrize("ctx_len",json_data['ctx_len'])
+        metafunc.parametrize("mxfp6",json_data['mxfp6'])
+        metafunc.parametrize("mxint8",json_data['mxint8'])
+        metafunc.parametrize("device_group",json_data['device_group'])
+  
+def pytest_collection_modifyitems(items):
+    """
+    pytest_collection_modifyitems is pytest a hook,
+    which is used to re-order the execution order of the testing script/methods 
+    with various combination of inputs. 
+    called after collection has been performed, may filter or re-order the items in-place.
+    Parameters:	
+    items (List[_pytest.nodes.Item]) list of item objects
+    ----------
+    Ref: https://docs.pytest.org/en/4.6.x/reference.html#collection-hooks
+    """
+    print("\n*************Initial Test script/functions execution order****************\n\n",items)
+    if len(items)>=4:
+        run_first = ["test_export","test_compile","test_execute","test_infer"]

```

Comment:
after "test_export", "test_compile", "test_execute", we should remove onnx,qpc of few models to correctly test infer.

Reply:
I am doing this for the first model, not for all the models.